docs/git.md

```diff
@@ -0,0 +1,87 @@
+# Simperby: the Git-based Blockchain
+
+One of the most unique features of Simperby is its use of Git.
+
+## Pre-requisites
+
+You should have a basic understanding of
+
+1. Blockchain
+2. [Simperby Consensus and Governance](./protocol_overview.md)
+3. Git
+
+## Summary
+
+1. **Every finalized data in a Simperby chain is stored in a Git repository.**
+2. It includes transactions, agendas, agenda proofs, chats, and blocks, linearly committed in the `main` branch.
+3. A Simperby node manages its own Git repository and provides a Git server of it to the node operator.
+4. The node operator can fetch the blockchain data, walk through the history, and check the diffs using the Git protocol.
+5. **Every to-be-finalized data is also managed in a Git repository.**
+6. All the pending agendas (waiting for approval) will be presented as multiple branches grown from the `main` branch.
+7. The node operator may create their own transaction as a commit and push to a particular branch which represents an agenda proposal.
+8. **Simperby funtions as a general distributed Git repository**, that may contain any useful data for the organization. This is trivially achieved because **we take transactions as Git commits.** (This can be understood as exploiting the 'blockchain state' as a Git repository)
+
+## Lifecycle of a Simperby Chain
+
+TODO
+
+### Step 0: finalized block
+
+Let's assume that there is the last finalized block with the height of `H`.
+We will take that has a starting point of our recursive process. Of course the base case of the finalized block would be the genesis block.
+
+### Step 1: Agenda
+
+TODO
+
+## Specification
+
+Here we present the specification of the Simperby Git repository.
+
+### Commits
+
+A commit is defined as either
+
+1. `initial`: an empty, initial commit as the very first commit of the repository.
+2. `genesis`: a commit that contains the initial state and the genesis info.
+3. `block`: a finalized block
+4. `tx`: a transaction of an arbitrary update on the state. Note that a `tx` commit is the only exception that the commit title does not start with its type, `tx`.
```

Comment:
Is it arbitrary update on the state?
Can this change the delegation state as well?

Reply:
Explained.